pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -236,3 +248,61 @@ def fixdict(d):
     projections = fixdict(projections)
     sps(projections)
     sr('ConnectLayers')
+
+
+def _check_input_nodes(pre, post, conn_spec):
+    '''
+    Check the properties of `pre` and `post` nodes:
+
+    * if `conn_spec` is "one_to_one", skip uniqueness check for better speed
+      and use data_connect
+    * if both `pre` and `post` are NodeCollections or can be converted to
+      NodeCollections (i.e. contain unique IDs), then proceed to "normal"
+      connect (potentially after conversion to NodeCollection)
+    * if both `pre` and `post` are arrays and contain non-unique items, then
+      we proceed to "data_connect"
+    * if at least one of them has non-unique items and they have different
+      sizes, then raise an error.
+    '''
+    data_connect = False
+
+    pre_is_nc, post_is_nc = True, True
+
+    if not isinstance(pre, NodeCollection):
+        # skip uniqueness check for data_connect compatible `conn_spec`
+        if conn_spec != 'one_to_one' and len(set(pre)) == len(pre):
+            pre = NodeCollection(pre)
+        else:
+            pre_is_nc = False
+
+    if not isinstance(post, NodeCollection):
+        # skip uniqueness check for data_connect compatible `conn_spec`
+        if conn_spec != 'one_to_one' and len(set(post)) == len(post):
+            post = NodeCollection(post)
+        else:
+            post_is_nc = False
+
+    if not pre_is_nc or not post_is_nc:
+        assert len(pre) == len(post), "If `pre` or `post` contain non-unique IDs, then " \
+                                      "they must have the same length."
+
+        # convert them to arrays
+        if not pre_is_nc and not isinstance(pre[0], (int, np.integer)):
+            raise NESTErrors.ArgumentType("Connect", " `pre` IDs should be integers.")
+
+        if not post_is_nc and not isinstance(post[0], (int, np.integer)):
+            raise NESTErrors.ArgumentType("Connect", " `post` IDs should be integers.")
+
+        pre = np.asarray(pre, dtype=int)
+        post = np.asarray(post, dtype=int)
```

Comment:
If you use implicit `dtype` in `np.asarray()`, you can check the `dtype` of the arrays afterwards to see if all IDs are integers.